### DISCUSSION TITLE
RTE Component isn't re-rendering

### DISCUSSION THREAD

**Discussion by suchintan** [mantinedev/mantine#2000] [Q&A] [answered]
Hi,

Is there a way to force the RTE to re-render itself? I would have expected that as the "value" prop changes that the RTE re-renders itself, but I don't see that happening.

I'm using react router and grabbing the "templateId" off the url

Here's my code:
```
  const { templateId } = useParams()
  const [templateContents, updateTemplateContents] = useState(
    templateId || "blabla"
  )
  useEffect(() => updateTemplateContents(templateId || "bla bla"), [templateId])
  const ref = useRef<Editor>(null)

  console.log(`TemplateId=${templateId}`)
  console.log(`TemplateContents=${templateContents}`)
  const focus = () => {
    if (ref.current != null) ref.current.focus()
  }
  return (
    <RichTextEditor
      ref={ref}
      value={templateContents}
      onChange={updateTemplateContents}
      onClick={focus}
      onImageUpload={handleImageUpload}
      sx={{ height: "100%", cursor: "text" }}
    />
  )
```

Initial state:
> url: /template/t_12351

<img width="553" alt="image" src="https://user-images.githubusercontent.com/3853670/182742189-a9175f31-399f-4aca-a065-4c8ea34deacb.png">

I change url by clicking on something in my app
> url: /template/t_12358

<img width="496" alt="image" src="https://user-images.githubusercontent.com/3853670/182742238-c41120c7-9a76-432a-8b3c-94bdcff69528.png">

I can see the variable defined by the state hook updating, but the RTE is not rendering the new contents. I'm not sure if it's a bug with RTE or if I'm doing something wrong

**Comment by rtivital** [comment, marked as answer]
See https://github.com/mantinedev/mantine/issues/1625#issuecomment-1154506482

**Reply by suchintan** [reply]
This worked!! Thank you so much :)
